src/platform/packages/shared/kbn-field-utils/src/components/field_select/utils.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { i18n } from '@kbn/i18n';
+import { KBN_FIELD_TYPES, esFieldTypeToKibanaFieldType } from '@kbn/field-types';
+import { getTextBasedColumnIconType } from '../../utils/get_text_based_column_icon_type';
+import { getFieldTypeName } from '../../utils/get_field_type_name';
+
+export const TYPE_GROUPS: { label: string; types: string[] }[] = [
```

Comment:
Is the getKbnFieldType already does something similar rather than having these hardcoded here?